crates/net/network/src/manager.rs

```diff
@@ -673,261 +659,276 @@ where
         // If the budget is exhausted we manually yield back control to the (coop) scheduler. This
         // manual yield point should prevent situations where polling appears to be frozen. See also <https://tokio.rs/blog/2020-04-preemption>
         // And tokio's docs on cooperative scheduling <https://docs.rs/tokio/latest/tokio/task/#cooperative-scheduling>
-        let mut budget = 1024;
-
-        loop {
-            // advance the swarm
-            match this.swarm.poll_next_unpin(cx) {
-                Poll::Pending | Poll::Ready(None) => break,
-                Poll::Ready(Some(event)) => {
-                    // handle event
-                    match event {
-                        SwarmEvent::ValidMessage { peer_id, message } => {
-                            this.on_peer_message(peer_id, message)
-                        }
-                        SwarmEvent::InvalidCapabilityMessage { peer_id, capabilities, message } => {
-                            this.on_invalid_message(peer_id, capabilities, message);
-                            this.metrics.invalid_messages_received.increment(1);
-                        }
-                        SwarmEvent::TcpListenerClosed { remote_addr } => {
-                            trace!(target: "net", ?remote_addr, "TCP listener closed.");
-                        }
-                        SwarmEvent::TcpListenerError(err) => {
-                            trace!(target: "net", ?err, "TCP connection error.");
-                        }
-                        SwarmEvent::IncomingTcpConnection { remote_addr, session_id } => {
-                            trace!(target: "net", ?session_id, ?remote_addr, "Incoming connection");
-                            this.metrics.total_incoming_connections.increment(1);
-                            this.metrics
-                                .incoming_connections
-                                .set(this.swarm.state().peers().num_inbound_connections() as f64);
-                        }
-                        SwarmEvent::OutgoingTcpConnection { remote_addr, peer_id } => {
-                            trace!(target: "net", ?remote_addr, ?peer_id, "Starting outbound connection.");
-                            this.metrics.total_outgoing_connections.increment(1);
-                            this.metrics
-                                .outgoing_connections
-                                .set(this.swarm.state().peers().num_outbound_connections() as f64);
+        // poll new block imports (dummy stream)
+        let maybe_more_block_imports = poll_nested_stream_with_yield_points!(
+            "net",
+            "Block imports stream",
+            DEFAULT_BUDGET_TRY_DRAIN_STREAM,
+            this.block_import.poll_next_unpin(cx),
+            |outcome| this.on_block_import_result(outcome),
+        );
```

Comment:
this is not necessary,

we don't even have block imports, and even if we had this wouldn't produce much

Reply:
yeah I was wondering why it was there